### DISCUSSION TITLE
Pagination listing invalid

### DISCUSSION THREAD

**Discussion by staticdreams** [livewire/livewire#4653] [7. Bugs (archived)] [answered]
Our database has a lot of records. Currently 5000+ but we expect it to be around 10k.
As as result we have more then 300 pages (15 records per page)

We're experiencing pagination issues when browsing through pages (especially when using navigation arrows : left and right)
The `filterCards()` is an Eloquent `Card::query()` with nothing special in it. Just a few where clauses and `orderBy` at the end.

Then I'm doing something like this in render method:

    public function render()
    {
        return view('livewire.cards.cards-list', [
            'cards' => $this->filterCards()->paginate(15)
        ]);
    }

As illustrated below, we're starting out as expected with 15 records and all pagination aligned correctly.
It goes smoothly until we hit page 10. 
That's were pagination starts to behave differently, breaking the pages order.
Clicking directly on each page number would list pages correctly, btw.
When using navigation arrows the pagination block will start to grow, breaking out of page layout.
Interestingly enough, when click on page 2 (last slide) would also leave the pagination broken, but when the page is refreshed (with /?page=2) - comes back to normal.

![pagination-bug](https://user-images.githubusercontent.com/988705/153582240-4cddfd6e-aa77-4b3e-bb03-f9a2aaea43aa.jpg)

Something tells me it is not intended behaviour of pagination and I haven't found a similar issue already submitted (I'm sorry if I have missing it). I'm using livewire v 2.5.5 and perhaps this has been already addressed ? (Laravel 8.12)

**Comment by satoved** [comment]
Are you using the WithPagination trait?

**Reply by staticdreams** [reply]
I sure do.

**Comment by joshhanley** [comment]
@staticdreams what version of Livewire are you using? Also do you publish your pagination views or use the built in ones? We released a fix for that issue somewhere around v2.6 (I think).

**Reply by staticdreams** [reply]
Hey Josh,

I managed to upgrade to the latest version of 2.10.2 and Laravel version 8.83 but unfortunately the problem seem to persist.
I do not publish the pagination views.

Anything else I can do in this situation?

**Comment by joshhanley** [comment]
@staticdreams hmm ok that is strange.

It looks like a morphdom error. Can you inspect one of the numbered page buttons in dev tools and take a screenshot of the rendered html?

**Reply by staticdreams** [reply]
Sure. 

![CleanShot 2022-02-16 at 13 35 02@2x](https://user-images.githubusercontent.com/988705/154265848-e678c0a6-6356-45b4-8f1c-10326b1f698c.jpg)

![CleanShot 2022-02-16 at 13 35 51@2x](https://user-images.githubusercontent.com/988705/154265896-16047a53-6737-4011-be90-b503b5c6505a.jpg)

**Reply by staticdreams** [reply]
One more another precision: if I load page normally (e.g loading url with ?page=17) the order in pagination is correct and visually it behaves as expected. However it all breaks when one of the button is clicked (either "prev" / "next" or page number)

**Reply by joshhanley** [reply]
@staticdreams how strange, it looks like the keys are coming through as expected, which is what should tell morphdom when to add/remove elements.

Do you happen to have two lots of pagination links on the same page (ie top and bottom of your results)?

**Reply by staticdreams** [reply]
@joshhanley you're absolutely right! Thank you so much!
I actually did have two lists and two navigations beneath the lists. 
One for desktop, another for mobile version. Visually they are different and also located within different parent components.
I tried removing the mobile one and everything worked!
Is there a way to keep the 2 together or I will have to hack the structure a bit to accommodate the the two listings to live with one pagination?

**Comment by joshhanley** [comment, marked as answer]
@staticdreams yep, we actually merge a fix in for that in #3631 which should add an auto incrementing value to your paginator to allow multiple sets of links on a page.

But looking at your render screenshots, it doesn't seem to be working. The keys should look like `paginator-1-page1` but yours are rendering like `paginator-page1`.

Maybe there is a bug with it. Are you using a custom paginator page name or just the default `page`?

One thing actually to try, it doing a `php artisan view:clear` and then re-checking the rendered html to see if it changes to what I described above.

**Reply by staticdreams** [reply]
@staticdreams Josh, thanks again!
 I think it's all good now. I cleared the view and remove the published assets. It seems to work with the two views now.
Cheers, mate!

**Reply by joshhanley** [reply]
@staticdreams no worries! 🙂
